### DISCUSSION TITLE
How to let StackPanel's children to fill maximum space

### DISCUSSION THREAD

**Discussion by ismlsmile** [AvaloniaUI/Avalonia#14674] [General]
Example:

	<TabControl>
		<TabControl.Items>
			<TabItem Header="Item 1">
				<StackPanel>
					<TextBlock Text="This is test" />
					<Border x:Name="testBorder" Background="Red" Height="200" />
				</StackPanel>
			</TabItem>
			<TabItem Header="Item 2">
				<TextBlock Text="Page 2" />
			</TabItem>
		</TabControl.Items>
	</TabControl>
</UserControl>

I want to let the control "testBorder" to fill maximum space, but it seems that avalonia does not support such syntax like Height="100%", how to do this ?

**Comment by gehongyan** [comment]
Setting `VerticalAlignment` or `HorizontalAlignment` to `Stretch` can make content elements occupy all available space within the container. However, the nature of a `StackPanel` allows its size to be considered infinite in the direction of its `Orientation`. Therefore, `Stretch` is ineffective. As a result, `StackPanel` may not meet your needs.

Consider using `Grid`, `Panel`, or other layout containers more suitable for your application scenario.

See:

[Layout | Avalonia Docs](https://docs.avaloniaui.net/docs/basics/user-interface/building-layouts/)
[Layout Controls | Avalonia Docs](https://docs.avaloniaui.net/docs/reference/controls/layout-controls)

---

Additionally, if you persist in using a `StackPanel` as the layout container, there may be a solution for dynamically binding the `Height`:

```csharp
// Declares converters supporting Math calculations
using Avalonia.Data.Converters;

namespace YourNamespace;

public static class MathConverters
{
    /// <summary>
    ///     Adds all the input values together.
    /// </summary>
    public static readonly IMultiValueConverter Add =
        new FuncMultiValueConverter<double, double>(x => x.Sum());

    /// <summary>
    ///     Negates the input value.
    /// </summary>
    public static readonly IValueConverter Negate =
        new FuncValueConverter<double, double>(x => -x);
}
```

```xaml
<!-- Add your namespace alias -->
<!-- xmlns:local="clr-namespace:YourNamespace" -->
<StackPanel>
    <TextBlock Name="headerTextBlock" Text="This is test" />
    <Border x:Name="testBorder" Background="Red">
        <Border.Height>
            <MultiBinding Converter="{x:Static local:MathConverters.Add}">
                <!-- Bounds property provides the actual size of an element for a Visual. -->
                <Binding Path="$parent[StackPanel].Bounds.Height" />
                <Binding Path="#headerTextBlock.Bounds.Height" Converter="{x:Static local:MathConverters.Negate}" />
            </MultiBinding>
        </Border.Height>
    </Border>
</StackPanel>
```

**Comment by ismlsmile** [comment]
I tried Grid, and it could meet my needs. Thanks for your answer.
